### PR TITLE
Fix type for pipeline resources for data factories

### PR DESCRIPTION
**Pull request by bgold09**
* Although the child resources are known publicly as "pipelines", the
    type name in the API route is "datapipelines".
  * This will fix resource creation and export via deployment template
    schemas.

### CODE REVIEW

File: schemas/2015-10-01/Microsoft.DataFactory.json [hunk 1]
```diff
@@ -141,7 +141,7 @@
           ]
         },
         "type": {
-          "enum": [ "Microsoft.DataFactory/datafactories/pipelines" ]
+          "enum": [ "Microsoft.DataFactory/datafactories/datapipelines" ]
```
Comment (wenwu449) [inline schemas/2015-10-01/Microsoft.DataFactory.json]: type should be exactly same with the word in service's manifest, including casing.

Comment (bgold09) [inline schemas/2015-10-01/Microsoft.DataFactory.json]: @wenwu449 should be fixed now, if I understood you correctly (changed line 128 to be "datapipelines")

File: schemas/2015-10-01/Microsoft.DataFactory.json [hunk 2]
```diff
@@ -125,7 +125,7 @@
         }
       }
     },
-    "pipelines": {
+    "dataPipelines": {
```
Comment (wenwu449) [inline schemas/2015-10-01/Microsoft.DataFactory.json]: you need to update the 2 deploymentTemplate.json files for this reference name change.

Comment (bgold09) [inline schemas/2015-10-01/Microsoft.DataFactory.json]: forgot that; fixed

**Pull request merged**